src/api.js

```diff
@@ -2455,7 +2452,52 @@ api.declare({
   return res.reply(Object.assign({}, worker.json(), {actions}));
 });
 
-// /** Update a worker */
+/** Quarantine a Worker */
+api.declare({
+  method: 'put',
+  route:  '/provisioners/:provisionerId/worker-types/:workerType/workers/:workerGroup/:workerId',
+  name:   'quarantineWorker',
+  stability: API.stability.experimental,
+  scopes: [
+    [
+      'queue:quarantine-worker:<provisionerId>/<workerType>/<workerGroup>/<workerId>',
+    ],
+  ],
+  deferAuth: true,
```

Comment:
I don't think this is necessary here -- all of the scopes' parameters are already available.